src/lib/configs/project/providerArtifacts.ts

```diff
@@ -151,18 +211,62 @@ const migrations: Migrations<Config> = [
 
     return { version: 1, offers };
   },
+  async (config: Config): Promise<ConfigV2> => {
+    if (!validateConfigSchemaV1(config)) {
+      throw new Error(
+        `Migration error. Errors: ${await validationErrorToString(
+          validateConfigSchemaV0.errors,
+        )}`,
+      );
+    }
+
+    const configPath = join(
+      getFluenceDir(),
+      PROVIDER_ARTIFACTS_CONFIG_FULL_FILE_NAME,
+    );
+
+    const newConfig: ConfigV2 = {
+      version: 2,
+      offers: {},
+    };
+
+    const { signerOrWallet } = await getDealClient();
+
+    for (const [env, configPerEnv] of Object.entries(config.offers)) {
+      if (!isFluenceEnv(env)) {
+        throw new Error(
+          `Unreachable. Migration error. Unknown env ${env} in ${PROVIDER_ARTIFACTS_CONFIG_FULL_FILE_NAME}`,
+        );
+      }
+
+      const newConfigPerEnv: Record<string, OfferConfigV2> = {};
+
+      for (const [offerName, offer] of Object.entries(configPerEnv)) {
+        const providerAddress = await input({
+          message: `Enter provider address that was used when creating offer ${offerName} with offerId: ${offer.id} at ${configPath}`,
```

Comment:
Provider address is not stored along with the offers? Maybe it should be?

Reply:
this IS the idea in this PR to now store the provider address with the offer
here is the new config type https://github.com/fluencelabs/cli/pull/906/files/181239fe1dea4b3f430e2b79b630e7165016b77e#diff-75e8e297e5ce3647b9b327aa39919ad1bd3d7783560bfc361dba61658c04b15eR122-R125